Classes/core/entity/tower/aggressive_magician.cpp

```diff
@@ -1,10 +1,46 @@
 #include "aggressive_magician.h"
 #include "../../map.h"
+#include <stdexcept>
 
 namespace towerdefence {
     namespace core {
+        AggressiveMagician::AggressiveMagician(id::Id id, const timer::Clock &clk)
+                : Tower(id), release_skill_(clk.with_period_sec(70)),
+                  attack_(clk.with_period_sec(INFO.attack_interval_)) {}
         void AggressiveMagician::on_tick(GridRef g) {
             Tower::on_tick(g);
+            auto status = this->status();
+            auto &clk = g.clock();
+            if (clk.is_triggered(this->attack_)){
+                this->attack_ = clk.never();
+                for (auto grid : g.with_radius(status.attack_radius_, linf_dis)) {
+                    grid.current().with_tower(
+                            [this, &clk = g.clock()](std::unique_ptr<Tower> &tower) {
+                                tower->add_buff({this->id, Buff::DEFAULT},
+                                                Buff::attack(0.20));
+                            });
+                }
+            }
+            if (clk.is_triggered(release_skill_)) {
+                this->release_skill_ = clk.never(); // stop release_skill
+                for (auto grid : g.with_radius(status.attack_radius_, linf_dis)) {
+                    grid.current().with_tower(
+                            [this, &clk = g.clock()](std::unique_ptr<Tower> &tower) {

```

Comment:
```suggestion
                            [this, &clk](std::unique_ptr<Tower> &tower) {

```
It's unnecessary to initialize a `clk`.